### PR TITLE
[Docs] Migration Guide of Datasets

### PR DESCRIPTION
**Pull request by xinke-wang**

### CODE REVIEW

File: docs/en/migration/dataset.md [hunk 1]
```diff
@@ -1 +1,254 @@
...
+      ...
+    ]
+}
+```
+
+Based on the above structure, we introduced `TextDetDataset`, `TextRecogDataset` for MMOCR-specific tasks.
+
+### Text Detection
+
+1. **Introduction of the New Format**
```
Comment (gaotongxiao) [inline docs/en/migration/dataset.md:93]: Consider using headings (####)

File: docs/en/migration/dataset.md [hunk 2]
```diff
@@ -1 +1,254 @@
...
+
+In consideration of the cost to users for data migration, we have temporarily made MMOCR version 1.x compatible with the old MMOCR 0.x format.
+
+```{note}
+The code and components used for compatibility with the old data format may be completely removed in a future release. Therefore, we strongly recommend that users migrate their datasets to the new data format.
+```
+
+Specifically, we provide three dataset classes [IcdarDataset](mmocr.datasets.IcdarDataset), [RecogTextDataset](mmocr.datasets.RecogTextDataset), [RecogLMDBDataset](mmocr.datasets.RecogLMDBDataset) to support the old formats.
+
+1. [IcdarDataset](mmocr.datasets.IcdarDataset) supports the COCO-like format annotations for text detection. The users just need to add a new dataset config to `configs/textdet/_base_/datasets` and specify its dataset type as `IcdarDataset`.
```
Comment (gaotongxiao) [inline docs/en/migration/dataset.md:192]:
```suggestion
1. [IcdarDataset](mmocr.datasets.IcdarDataset) supports COCO-like format annotations for text detection. You just need to add a new dataset config to `configs/textdet/_base_/datasets` and specify its dataset type as `IcdarDataset`.
```

File: docs/en/migration/dataset.md [hunk 3]
```diff
@@ -1 +1,254 @@
...
+   train_dataset = dict(
+       type='IcdarDataset',
+       data_root=data_root,
+       ann_file=train_anno_path,
+       data_prefix=dict(img_path='imgs/'),
+       filter_cfg=dict(filter_empty_gt=True, min_size=32),
+       pipeline=None)
+   ```
+
+2. [RecogTextDataset](mmocr.datasets.RecogTextDataset) supports `.txt` and `.jsonl` format annotations for text recognition. The users just need to add a new dataset config to `configs/textrecog/_base_/datasets` and specify its dataset type as `RecogTextDataset`. For example, the following example shows how to configure and load the 0.x format labels `old_label.txt` and `old_label.jsonl` from the toy dataset.
```
Comment (gaotongxiao) [inline docs/en/migration/dataset.md:207]:
```suggestion
2. [RecogTextDataset](mmocr.datasets.RecogTextDataset) supports `.txt` and `.jsonl` format annotations for text recognition. You just need to add a new dataset config to `configs/textrecog/_base_/datasets` and specify its dataset type as `RecogTextDataset`. For example, the following example shows how to configure and load the 0.x format labels `old_label.txt` and `old_label.jsonl` from the toy dataset.
```

File: docs/en/migration/dataset.md [hunk 4]
```diff
@@ -1 +1,254 @@
...
+        data_root=data_root,
+        ann_file='old_label.jsonl',
+        data_prefix=dict(img_path='imgs'),
+        parser_cfg=dict(
+            type='LineJsonParser',
+            keys=['filename', 'text'],
+        pipeline=[])
+   ```
+
+3. [RecogLMDBDataset](mmocr.datasets.RecogLMDBDataset) supports LMDB format annotations for text recognition. The users just need to add a new dataset config to `configs/textrecog/_base_/datasets` and specify its dataset type as `RecogLMDBDataset`. For example, the following example shows how to configure and load the **label-only lmdb** `label.lmdb` from the toy dataset.
```
Comment (gaotongxiao) [inline docs/en/migration/dataset.md:236]:
```suggestion
3. [RecogLMDBDataset](mmocr.datasets.RecogLMDBDataset) supports LMDB format annotations for text recognition. You just need to add a new dataset config to `configs/textrecog/_base_/datasets` and specify its dataset type as `RecogLMDBDataset`. For example, the following example shows how to configure and load the **label-only lmdb** `label.lmdb` from the toy dataset.
```

File: docs/en/migration/dataset.md [hunk 5]
```diff
@@ -1 +1,254 @@
...
+        data_prefix=dict(img_path='imgs'),
+        pipeline=[])
+   ```
+
+   When the `lmdb` file contains **both labels and images**, in addition to setting the dataset type to `RecogLMDBDataset` as in the above example, we also need to configure the file backend in `file_client_args` to `lmdb` in the data pipelines and specify `db_path` as the path to the `lmdb` annotation file as in the following example.
+
+   ```python
+    # Set the file backend to lmdb and specify the db_path
+    file_client_args = dict(backend='lmdb', db_path='tests/data/rec_toy_dataset/imgs.lmdb')
+   ```
```
Comment (gaotongxiao) [inline docs/en/migration/dataset.md:254]: Use [`LoadImageFromLMDB`](https://github.com/open-mmlab/mmocr/blob/e760dcd1dd89d6c136db2bec8c56bab00109d201/mmocr/datasets/transforms/loading.py#L445)

**Review by gaotongxiao — APPROVED** [review]
